test-data/unit/check-classes.test

```diff
@@ -2055,3 +2055,107 @@ class B(object, A): # E: Cannot determine consistent method resolution order (MR
 # flags: --fast-parser
 class C(metaclass=int()):  # E: Dynamic metaclass not supported for 'C'
     pass
+
+[case testVariableSubclass]
+class A:
+    a = 1 # type: int
+class B(A):
+    a = None
```

Comment:
Hm, this would become invalid with strict none checking. Maybe just use another int value?